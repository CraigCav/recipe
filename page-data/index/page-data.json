{"componentChunkName":"component---src-pages-index-js","path":"/","webpackCompilationHash":"356c1bd9f0f7a983bfed","result":{"pageContext":{"isCreatedByStatefulCreatePages":true}}}