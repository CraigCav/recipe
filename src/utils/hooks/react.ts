import React from 'react';

let currentInstance = null;

export const use = <P>(render: React.SFC<P>): React.ComponentClass<P> => {
  return class extends React.Component<P> {
    state = {};
    effects = [];
    componentDidMount() {
      this.afterRender();
    }
    componentDidUpdate() {
      this.afterRender();
    }
    componentWillUnmount() {
      this.effects.forEach(({destroy}) => {
        if (destroy) destroy();
      });
    }
    afterRender() {
      this.effects.forEach(({create, destroy}) => {
        if (destroy) destroy();
        create();
      });
    }
    render() {
      currentInstance = this;
      Object.assign(currentInstance, {callIndex: 0, effects: []});
      return render(this.props);
    }
  };
};

const nextStateCell = () => {
  const frame = currentInstance;
  const {state, callIndex} = currentInstance;
  currentInstance.callIndex++;
  return {
    frame,
    state,
    key: callIndex,
  };
};

export const useRef = <TValue>(initialValue: TValue): {current: TValue} => {
  const {state, key} = nextStateCell();

  if (!state.hasOwnProperty(key)) {
    state[key] = {current: initialValue};
  }

  return state[key];
};

export const useState = <TValue>(initialValue: TValue): [TValue, (value: TValue) => void] => {
  const {frame, state, key} = nextStateCell();

  return [
    state.hasOwnProperty(key) ? state[key] : initialValue,
    newValue => frame.setState({[key]: newValue}),
  ];
};

const inputsAreEqual = (next, prev) => next && next.every((d, i) => d === prev[i]);

export const useEffect = (create, inputs = [create]) => {
  const {frame, state, key} = nextStateCell();

  const prevEffect = state[key];

  if (prevEffect && inputsAreEqual(inputs, prevEffect.inputs)) return;

  const effect = (state[key] = {
    create,
    inputs,
    destroy: prevEffect ? prevEffect.destroy : null,
  });

  frame.effects.push(effect);
};
