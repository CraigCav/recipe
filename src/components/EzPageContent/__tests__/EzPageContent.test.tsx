import React from 'react';
import {axe} from 'jest-axe';
import {visualSnapshots} from 'sosia';
import {render} from '@testing-library/react';
import markdown from '../EzPageContent.md';
import {EzPageContent, EzContentGroup} from '..';
import {EzCard, EzSegmentedControl} from '../../index';

const scope = {EzPageContent, EzCard, EzContentGroup, EzSegmentedControl};

describe('EzPageContent', () => {
  visualSnapshots({markdown, scope});

  it('should meet accessibility guidelines', async () => {
    const {container} = render(
      <EzPageContent>
        <EzContentGroup>
          <p>Lorem ipsum dolor</p>
          <p>Lorem ipsum dolor</p>
        </EzContentGroup>
      </EzPageContent>
    );
    const actual = await axe(container.outerHTML);
    expect(actual).toHaveNoViolations();
  });
});
