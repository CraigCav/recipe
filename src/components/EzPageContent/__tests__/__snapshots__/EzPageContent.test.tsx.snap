// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EzPageContent should render EzContentGroup content with horizontal styles 1`] = `
.recipe-2 {
  background: #f4f7f8;
  padding: 0.8571428571428571rem 0.5714285714285714rem;
}

@media screen and (min-width:48em) {
  .recipe-2 {
    padding: 2.2857142857142856rem;
  }
}

.recipe-2 > *:not(:first-child) {
  margin-top: 2.2857142857142856rem;
}

.recipe-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.recipe-0 > * {
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.recipe-0 > *:not(:first-child) {
  margin-left: 1.4285714285714286rem;
}

<div
  className="recipe-2 recipe-3"
>
  <div
    className="recipe-0 recipe-1"
  >
    <p>
      Lorem ipsum dolor
    </p>
    <p>
      Lorem ipsum dolor
    </p>
  </div>
</div>
`;

exports[`EzPageContent should render EzContentGroup with default styles 1`] = `
.recipe-2 {
  background: #f4f7f8;
  padding: 0.8571428571428571rem 0.5714285714285714rem;
}

@media screen and (min-width:48em) {
  .recipe-2 {
    padding: 2.2857142857142856rem;
  }
}

.recipe-2 > *:not(:first-child) {
  margin-top: 2.2857142857142856rem;
}

.recipe-0 > *:not(:first-child) {
  margin-top: 1.1428571428571428rem;
}

<div
  className="recipe-2 recipe-3"
>
  <div
    className="recipe-0 recipe-1"
  >
    <p>
      Lorem ipsum dolor
    </p>
    <p>
      Lorem ipsum dolor
    </p>
  </div>
</div>
`;

exports[`EzPageContent should render with default styles 1`] = `
.recipe-0 {
  background: #f4f7f8;
  padding: 0.8571428571428571rem 0.5714285714285714rem;
}

@media screen and (min-width:48em) {
  .recipe-0 {
    padding: 2.2857142857142856rem;
  }
}

.recipe-0 > *:not(:first-child) {
  margin-top: 2.2857142857142856rem;
}

<div
  className="recipe-0 recipe-1"
>
  <p>
    Lorem ipsum dolor
  </p>
  <p>
    Lorem ipsum dolor
  </p>
</div>
`;
